lms/djangoapps/lms_xblock/runtime.py

```diff
@@ -190,6 +189,15 @@ def set_tag(self, scope, key, value):
         )
 
 
+class SettingsService(object):
+    """
+    Allow XBlocks to access the server settings
+    """
+    def get(self, setting_name, default=None):
+        """ Gets setting value from django settings """
+        return getattr(settings, setting_name, default)
```

Comment:
@cpennington of course it could. The question, however, is whether it would be a whitelist or a blacklist.

Just to make sure we're on the same page, whitelist would mean there would be a list of settings this service is allowed access to; anything not on the list is restricted.

Whitelist simplifies adding new settings with restricted access; it's more secure as any setting is restricted by default. On the other hand, the service would need to know what settings are allowed, so it is likely to contain some data specific to particular XBlocks.

Blacklist approach is just a reverse of whitelist, with pros and cons swapped.

I would suggest going whitelist, so just letting you know that there's a certain tradeoff.